### ISSUE TITLE
Commercially-friendly licence?

### ISSUE THREAD

**Issue by franslundberg** [ianopolous/tweetnacl-java#6]
Hello! We (ASSA ABLOY AB) are using your library in-house for a demo. However, now we are considering making a product and we would have to replace your library due to licensing issues. Would you (the authors of the library) consider releasing it with an MIT license instead of GPL?  tweet-nacl is a low-level library, so the GLP license will be problematic for many projects, I guess.

Actually, I planned to release a secure channel implementation,
https://github.com/assaabloy-ppi/salt-channel, but I cannot use tweet-nacl because of
the GPL license. We want to release salt-channel with the MIT license.

Anyhow, I think there is some mutual benefits here. We could, for example, provide a real 
security audit. Please contact me at: frans.lundberg AT assabloy.com.

**Comment by ianopolous** [comment]
Yep, I'm fine with that. Crypto probably should be commercial friendly. I'll update the licence this week.

**Comment by franslundberg** [comment]
Great, that means we can contribute more to open source :-). Thank you for the quick reply.

**Comment by joakimeriksson** [comment]
Yes that sounds like an excellent step! In fact it should probably use the same licens of the initial tweet-nacl.c library that is included in this repo. At least the original license file should be included in the c-parts that are the same? And the port seems to be a very close C to Java port so there should be no reason to change the license at all from the MIT style that was used originally.

**Comment by boazy** [comment]
@ianopolous is there any update on the license? Java is sorely missing a pure JVM implementation of NaCl/TweetNaCl.

**Comment by jamesjb** [comment]
+1 on a commercial friendly license for this library, as I also have a project that would use this if it had an MIT-style license instead of GPL.

**Comment by raed667** [comment]
Hello, 
is dual-licensing this lib is still considered, or should I look elsewhere?

**Comment by zean00** [comment]
Hi,

Is there any update on the new license?
